default/python/AI/ResourcesAI.py

```diff
@@ -84,6 +96,39 @@ def bake_future_focus(self, pid, focus, update=True):
         Return success or failure
         """
         pinfo = self.raw_planet_info.get(pid)
+        if (focus == INDUSTRY or focus == RESEARCH) and not force:
+            idx = 0 if focus == INDUSTRY else 1
+            # check for influence instead
+            debug("possible_output of %s: %s", pinfo.planet.name, str(pinfo.possible_output))
+            focus_gain = pinfo.possible_output[focus][idx] - pinfo.possible_output[INFLUENCE][idx]
+            influence_gain = pinfo.possible_output[INFLUENCE][2] - pinfo.possible_output[focus][2]
+            if influence_gain * self.priority[2] > focus_gain * self.priority[idx]:
+                debug(
+                    "Chosing influence over %s. %.2f * %.1f > %.2f * %.1f",
```

Comment:
Let's use f-string here. 

Float formatting looks lie that:
```python
a = 0.3333
f"{a:.2}"
'0.33'
```

Reply:
Hey, that looks almost like jinja, the one thing I like about ansible ;-)
But we need an extra f:
```
a=123.456
f"{a:.2}"
'1.2e+01'
f"{a:.2f}"
'123.46'
```

Reply:
> Hey, that looks almost like jinja, the one thing I like about ansible ;-)
 But we need an extra f:

If you do Jinja2 directly it requires a bit more than f.  Same for `c-style` and `.format`. 

To read this entry I need to go back of forth to the variable list.  `%s. %.2f * %.1f > %.2f * %.1f`, f string allows to read it in sequence.